apps/dash-lunar-lander/app.py

```diff
@@ -0,0 +1,435 @@
+import numpy as np
+import casadi
+from casadi import SX, DM
+from math import cos, sin
+import plotly.graph_objects as go
+
+app = dash.Dash(__name__)
+server = app.server
+
+app.layout = html.Div(
+    children=[
+        html.Div(
+            children=[
+                html.Img(src=app.get_asset_url("dash-logo.png"), id="logo"),
+                html.H4("Dash Lunar Lander"),
+                html.A(
+                    id="github-link",
+                    children=["View source code on Github"],
+                    href="https://github.com/plotly/dash-sample-apps/blob/master/apps/dash-lunar-lander/app.py",
+                ),
+            ],
+            className="app-header",
+        ),
+        html.Div(
+            [
+                html.H3("Lunar Lander Trade Study App"),
+                html.P(
+                    "This app allows for exploring the trade space of a lunar lander. You can adjust different "
+                    "parameters on the lander, like max angular velocity and inital mass, using the sliders and a new "
+                    "optimal trajectory will be recomputed in near real time using Casadi."
+                ),
+                html.A(
+                    "The lunar lander model was based off of one in this paper,",
+                    href="https://arxiv.org/pdf/1610.08668.pdf",
+                    target="_blank",
+                ),
+                html.P(
+                    "The lunar lander model does not allowed for free rotation, instead requiring the lander to "
+                    "gimbal it’s engine and thrust to impart a rotation, giving the mass optimal control case a "
+                    "distinctive hooked shape. Switching the optimizer to time optimal control results in a much "
+                    "smoother and more expected shape."
+                ),
+            ]
+        ),
+        # Display the Trajectory
+        dcc.Graph(id="indicator-graphic"),
+        # Adjust the Spacecraft Parameters
+        dcc.Slider(
+            id="m0Slider",
+            min=5000,
+            max=15000,
+            value=10000,
+            step=10,
+            marks={5000: {"label": "Inital Mass"}},
+        ),
+        dcc.Slider(
+            id="c1Slider",
+            min=35000,
+            max=44000 * 1.1,
+            value=44000,
+            step=10,
+            marks={35000: {"label": "Max Thrust"}},
+        ),
+        dcc.Slider(
+            id="isp", min=300, max=330, value=311, step=1, marks={300: {"label": "ISP"}}
+        ),
+        dcc.Slider(
+            id="c3Slider",
+            min=0.01,
+            max=0.1,
+            value=0.0698,
+            step=0.0001,
+            marks={0.01: {"label": "Max Angular Velocity"}},
+        ),
+        dcc.Slider(
+            id="gSlider",
+            min=0.5,
+            max=2,
+            value=1.6229,
+            step=0.0001,
+            marks={0.5: {"label": "Gravity"}},
+        ),
```

Comment:
![Screen Shot 2020-01-27 at 1 01 48 PM](https://user-images.githubusercontent.com/30911511/73200843-9d6c8c00-4105-11ea-8c99-0e912547a0c8.png)

Do you think we could align the slider labels by their left side? I'm not a fan how the "Max Angular Velocity" label sticks out so much

Reply:
good catch, will resolve in next commit